custom_components/rpi_gpio/__init__.py

```diff
@@ -44,19 +44,20 @@ def prepare_gpio(event):
 
 def update_gpiod_lines():
     global gpiod_config, gpiod_lines
-
+    
     if gpiod_lines:
-        gpiod_lines.__exit__()
+        gpiod_lines.__exit__(None,None,None)
 
     gpiod_lines = gpiod.request_lines(
-        "/dev/gpiochip0",
+        "/dev/gpiochip4",
```

Comment:
Can we find a way to autodetect whether 0 or 4 would be the correct default here?

Reply:
With gpiodetect is possible. Probably libgpiod offers a way for checking it. Let me have a look this afternoon (busy now) and will update my PR

Reply:
Updated my PR. No error handling there, but now it calculates the gpiochip device to use. Tested in Raspberry Pi 3 and 5